modules/accumulo/src/main/java/org/apache/fluo/accumulo/util/ByteArrayUtil.java

```diff
@@ -98,21 +96,59 @@ public static long decodeLong(byte[] ba) {
    * @param listOfBytes Bytes objects to concatenate
    * @return Bytes
    */
-  public static final byte[] concat(Bytes... listOfBytes) {
-    try {
-      // TODO calculate exact array size needed
-      ByteArrayOutputStream baos = new ByteArrayOutputStream();
-      DataOutputStream dos = new DataOutputStream(baos);
-
-      for (Bytes b : listOfBytes) {
-        WritableUtils.writeVInt(dos, b.length());
-        b.writeTo(dos);
-      }
+  public static byte[] concat(Bytes... listOfBytes) {
+    int offset = 0;
+    int counter = 0;
 
-      dos.close();
-      return baos.toByteArray();
-    } catch (IOException e) {
-      throw new RuntimeException(e);
+    for (Bytes b : listOfBytes) {
+      counter += b.length() + 1;
+    }
+
+    byte[] data = new byte[counter];
+    for (Bytes b : listOfBytes) {
+      writeVint(data, offset, b.length());
+      offset++;
```

Comment:
`writeVint` could return offset, then could write `offset = writeVint(data, offset, b.length())` instead of `offset++`